contracts/stellar-interchain-token-service/src/contract.rs

```diff
@@ -294,21 +292,27 @@ impl InterchainTokenServiceInterface for InterchainTokenService {
     #[when_not_paused]
     fn register_custom_token(
         env: &Env,
+        caller: Address,
         salt: BytesN<32>,
         token_address: Address,
         token_manager_type: TokenManagerType,
     ) -> Result<BytesN<32>, ContractError> {
+        caller.require_auth();
+
         // Custom token managers can't be deployed with native interchain token type, which is reserved for interchain tokens
         ensure!(
             token_manager_type != TokenManagerType::NativeInterchainToken,
             ContractError::InvalidTokenManagerType
         );
-        let token_id = Self::interchain_token_id(env, Address::zero(env), salt);
 
-        Self::ensure_token_not_registered(env, token_id.clone())?;
+        // Validates the token address and it's associated token metadata
+        let _ =
+            token_metadata::token_metadata(env, &token_address, &Self::native_token_address(env))?;
+
+        let token_id = Self::interchain_token_id(env, caller, salt);
```

Comment:
Still incorrect. See how ITS factory implements custom token linking, and derives the token id. You can compare it to the canonical token id derivation. Looking at ITS contract isn't sufficient, ITS factory is how the custom token linking is initiated.

https://github.com/axelarnetwork/interchain-token-service/blob/main/contracts/InterchainTokenFactory.sol#L500